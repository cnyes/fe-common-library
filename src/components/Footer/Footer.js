/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import React, { PureComponent, PropTypes } from 'react';
import cx from 'classnames';
import getStyleName from '../../utils/getStyleName';
import styles from './Footer.scss';
import footerLinks from './links';

class Footer extends PureComponent {
  static propTypes = {
    now: PropTypes.number.isRequired,
    Link: PropTypes.func,
  };

  static defaultProps = {
    Link: undefined,
  };

  static renderNavs() {
    return footerLinks.navs.map(item => (
      <a href={item.url} key={`footer-nav-${item.name}`} target="_blank" rel="noopener noreferrer">
        {item.title}
      </a>
    ));
  }

  static renderSocials() {
    return footerLinks.socials.map(item => (
      <div className={getStyleName(styles, 'social-item')} key={`footer-socials-${item.name}`}>
        <div className={getStyleName(styles, 'title')}>{item.title}</div>
        <a
          className={`${getStyleName(styles, 'social-icon')} ${getStyleName(styles, `cnyes-media-${item.name}`)}`}
          href={item.url}
          rel="noopener noreferrer"
          target="_blank"
        >
          {item.title}
        </a>
      </div>
    ));
  }

  render() {
    const { now, Link } = this.props;
    const thisYear = new Date(now).getFullYear();

    /* eslint-disable jsx-a11y/accessible-emoji */
    return (
      <div id={getStyleName(styles, 'cnyes-footer-wrapper')} className={cx('theme-footer-wrapper')}>
        <footer className={getStyleName(styles, 'main-footer')}>
          <div className={getStyleName(styles, 'logo')}>{Link ? <Link to="/" /> : <a href="/" />}</div>
          <div className={getStyleName(styles, 'nav')}>
            <nav>{this.constructor.renderNavs()}</nav>
            <small className={getStyleName(styles, 'copyright')}>
              © Copyright 2000-{thisYear} cnYES.com All rights reserved. 未經授權 不得轉載
            </small>
            <span className={getStyleName(styles, 'statement')}>
              <a href="https://news.cnyes.com/news/id/3919817" target="_blank" rel="noopener noreferrer">
                鉅亨網重大聲明
              </a>
            </span>
          </div>
          <div className={getStyleName(styles, 'socials')}>{this.constructor.renderSocials()}</div>
        </footer>
      </div>
    );
  }
}

export default Footer;
