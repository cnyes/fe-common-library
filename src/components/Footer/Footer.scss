@import '../../styles/base';

$footer-height: 140px;
$footer-bg-color: #514848;

#wrapper { // sass-lint:disable-line no-ids
  background-color: $footer-bg-color;
  height: $footer-height;

  /* NOTE: reset for export to common component start */
  font-smooth: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font: 400 17px/23px $font-family;

  * {
    box-sizing: border-box;
  }
  /* NOTE: reset for export to common component start end */

  .main-footer {
    background-color: $footer-bg-color;
    color: $white;
    display: table;
    margin: 0 auto;
    padding: 53px 0 40px;
    width: 890px;
  }

  .logo {
    display: table-cell;
    width: 168px;
    position: relative;

    a {
      font: inherit;
      background-size: cover;
      background: url('../../assets/icon-new-logo.svg') center no-repeat;
      position: absolute;
      left: 0;
      bottom: 0;
      display: inline-block;
      height: 56px;
      width: 128px;
    }
  }

  .nav {
    display: table-cell;
    font-size: 13px;
    line-height: 18px;
    padding-left: 40px;
    position: relative;
    width: 530px;

    &::before {
      border-left: 1px solid $white;
      content: '';
      height: 40px;
      left: 0;
      position: absolute;
      width: 1px;
    }

    a {
      color: $white;
      text-align: left;
      font: inherit;

      &:not(:first-child) {
        border-left: 1px solid $white;
        padding: 0 9px;
      }

      &:first-child {
        padding-right: 9px;
      }
    }
  }

  .socials {
    display: table-cell;
    white-space: nowrap;

    .social-item {
      display: inline-block;
      margin: 0 10px;
      position: relative;
    }

    .title {
      font-size: 12px;
      position: absolute;
      top: -13px;
      width: 61px;
      left: -7px;
      height: 13px;
      line-height: 13px;
      text-align: justify;
      transform: scale(0.75);

      &::after {
        content: '';
        display: inline-block;
        width: 100%;
      }
    }

    .social-icon {
      background-size: 44px 44px;
      display: inline-block;
      height: 44px;
      margin-top: 3px;
      width: 44px;
      text-indent: -5000px; /* hide inner text */
    }

    .facebook {
      background: url('../../assets/icon-white-bg-fb.svg') center no-repeat;
    }

    .line {
      background: url('../../assets/icon-white-line.svg') center no-repeat;
    }

    .app {
      background: url('../../assets/icon-download-app-qrcode.svg') center no-repeat;
    }
  }

  .copyright {
    font: inherit;
    display: block;
    margin-top: 3px;
  }

  .statement {
    bottom: -20px;
    position: absolute;
  }
}
