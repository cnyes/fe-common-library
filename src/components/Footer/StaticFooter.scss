@import '../../styles/base';

#cnyes-footer-wrapper { // sass-lint:disable-line no-ids
  /* NOTE: reset for export to common component start */
  font-smooth: auto;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font: 400 17px/23px $font-family;
  text-align: left;

  * {
    box-sizing: border-box;
  }
  /* NOTE: reset for export to common component start end */

  /* NOTE: handle for unexpected newline space start */
  .nav {
    nav{
      font-size: 0;

      a {
        font-size: 13px;
      }
    }
  }

  .socials {
    font-size: 0;
  }
  /* NOTE: handle for unexpected newline space end */
}
