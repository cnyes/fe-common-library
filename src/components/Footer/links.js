const footerLinks = {
  navs: [
    {
      title: '關於我們',
      name: 'about',
      url: 'http://www.cnyes.com/cnyes_about/cnyes_about.html',
    },
    {
      title: '廣告服務',
      name: 'ad',
      url: 'http://www.cnyes.com/cnyes_about/cnyes_AD01.html',
    },
    {
      title: '金融資訊元件',
      name: 'financial',
      url: 'http://www.cnyes.com/cnyes_about/cnyes_pas01.html',
    },
    {
      title: '聯絡我們',
      name: 'contact',
      url: 'http://www.cnyes.com/cnyes_about/cnyes_ctcUsTpe.html',
    },
    {
      title: '徵才',
      name: 'job',
      url: 'http://www.104.com.tw/jobbank/custjob/index.php?r=cust&j=5e6042253446402330683b1d1d1d1d5f2443a363189j01',
    },
    {
      title: '網站地圖',
      name: 'sitemap',
      url: 'http://www.cnyes.com/cnyes_about/site_map.html',
    },
    {
      title: '法律聲明',
      name: 'legal',
      url: 'http://www.cnyes.com/cnyes_about/cnyes_sos01.html',
    },
  ],
  socials: [
    {
      title: '粉 絲 團',
      name: 'facebook',
      url: 'https://www.facebook.com/cnYES/',
    },
    {
      title: '鉅亨網Line',
      name: 'line',
      url: 'https://line.me/ti/p/@ZLU0489G',
    },
    {
      title: '鉅亨網APP',
      name: 'app',
      url: 'http://www.cnyes.com/app?utm_source=cnyes&utm_medium=desktop&utm_campaign=desktop_footer',
    },
  ],
};

export default footerLinks;
