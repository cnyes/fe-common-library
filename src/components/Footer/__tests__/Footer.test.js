/* eslint-disable import/first,func-names,global-require,prefer-arrow-callback */
import React from 'react';
import { mount, shallow } from 'enzyme';
import Footer from '../Footer';

describe('Footer component test', () => {
  let Link;
  let now;

  beforeEach(() => {
    jest.resetModules();

    Link = jest.fn(() => <div className="mock-link" />);
    now = +new Date(2016, 9, 1);
  });

  it('should be same as snapshot', () => {
    const wrapper = mount(<Footer now={now} />);
    const tree = wrapper.html();

    expect(tree).toMatchSnapshot();
  });

  it('should has 11 anchor tags if props.Link is defined', () => {
    const wrapper = shallow(<Footer now={now} Link={Link} />);

    expect(wrapper.find('a').length).toEqual(11);
    expect(wrapper.html()).toContain('mock-link');
  });

  it('should has 12 anchor tags if props.Link is undefined', () => {
    const wrapper = shallow(<Footer now={now} />);

    expect(wrapper.find('a').length).toEqual(12);
    expect(wrapper.find('.mock-link').length).toEqual(0);
  });

  it('should has a <footer> tag which has class named `main-footer`', () => {
    const wrapper = shallow(<Footer now={now} />);

    expect(
      wrapper
        .find('footer')
        .first()
        .hasClass('main-footer')
    ).toBeTruthy();
  });
});
