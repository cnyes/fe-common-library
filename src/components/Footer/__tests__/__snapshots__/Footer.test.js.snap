// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test should be same as snapshot 1`] = `"<div id=\\"cnyes-footer-wrapper\\" class=\\"theme-footer-wrapper\\"><footer class=\\"main-footer\\"><div class=\\"logo\\"><!-- react-empty: 4 --></div><div class=\\"nav\\"><nav><a href=\\"http://www.cnyes.com/cnyes_about/cnyes_about.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">關於我們</a><a href=\\"http://www.cnyes.com/cnyes_about/cnyes_AD01.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">廣告服務</a><a href=\\"http://www.cnyes.com/cnyes_about/cnyes_pas01.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">金融資訊元件</a><a href=\\"http://www.cnyes.com/cnyes_about/cnyes_ctcUsTpe.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">聯絡我們</a><a href=\\"http://www.104.com.tw/jobbank/custjob/index.php?r=cust&amp;j=5e6042253446402330683b1d1d1d1d5f2443a363189j01\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">徵才</a><a href=\\"http://www.cnyes.com/cnyes_about/site_map.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">網站地圖</a><a href=\\"http://www.cnyes.com/cnyes_about/cnyes_sos01.html\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">法律聲明</a></nav><small class=\\"copyright\\"><!-- react-text: 15 -->© Copyright 2000-<!-- /react-text --><!-- react-text: 16 -->2016<!-- /react-text --><!-- react-text: 17 --> cnYES.com All rights reserved. 未經授權 不得轉載<!-- /react-text --></small><span class=\\"statement\\"><a href=\\"https://news.cnyes.com/news/id/3919817\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">鉅亨網重大聲明</a></span></div><div class=\\"socials\\"><div class=\\"social-item\\"><div class=\\"title\\">粉 絲 團</div><a class=\\"social-icon facebook\\" href=\\"https://www.facebook.com/cnYES/\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">粉 絲 團</a></div><div class=\\"social-item\\"><div class=\\"title\\">鉅亨網Line</div><a class=\\"social-icon line\\" href=\\"https://line.me/ti/p/@ZLU0489G\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">鉅亨網Line</a></div><div class=\\"social-item\\"><div class=\\"title\\">鉅亨網APP</div><a class=\\"social-icon app\\" href=\\"http://www.cnyes.com/app?utm_source=cnyes&amp;utm_medium=desktop&amp;utm_campaign=desktop_footer\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\">鉅亨網APP</a></div></div></footer></div>"`;
